### ISSUE TITLE
Extend unnecessary_unwrap to look for `expect` in addition to `unwrap`

### ISSUE THREAD

**Issue by shepmaster** [rust-lang/rust-clippy#7581]
### What it does

Applies the same logic as `unwrap` to `expect`.

```rust
fn main() {
    let t = Some(0);
    if t.is_some() {
        // let _v = t.unwrap(); // reported by clippy::unnecessary_unwrap
        let _v = t.expect("x"); // not reported
    }
}
```

### Categories

- Kind: complexity

### Drawbacks

None.

### Example

```rust
fn main() {
    let t = Some(0);
    if t.is_some() {
        let _v = t.expect("x");
    }
}
```

Could be written as:

```rust
fn main() {
    let t = Some(0);
    if let Some(_v) = t {}
}
```
